charts/zenko-zookeeper/templates/statefulset.yaml

```diff
@@ -43,7 +43,19 @@ spec:
         - name : ZOO_REPLICAS
           value: "{{ .Values.servers }}"
         - name : ZOO_HEAP_SIZE
-          value: "{{ .Values.heap }}"
+        {{- if .Values.resources.requests.memory }}
+          valueFrom:
+            resourceFieldRef:
+              containerName: {{ template "zookeeper.name" . }}-server
+              resource: requests.memory
```

Comment:
@NicolasT So this might work out better because the output is actually in bytes so we wouldn't even need the sed conversion

Reply:
Did you check whether `requests.memory` is not set (implicitly) when only `limits.memory` is set? Could you also check the value of the field ref when neither of them are set?

The rationale would be as follows: if the value (when neither are set) is an empty string, or whatever, then `ZOO_HEAP_SIZE` would be set to the empty string, in which case the default value in `kubernetes-entrypoint.sh` would be chosen and there's no need to have any if-else-else block here.

Reply:
Then, if the value is in bytes, or empty, we can simply pass it as-is to the JVM.

Reply:
So I just tested and `requests.memory` does seems to be set implicitly when only `limits.memory` is configured or if neither are set then the empty string is passed which sets the 2G default just as you expected.